client/gutenberg/extensions/giphy/edit.js

```diff
@@ -0,0 +1,193 @@
+		// If search is hardcoded Giphy URL following this pattern: https://giphy.com/embed/4ZFekt94LMhNK
+		if ( searchText.indexOf( '//giphy.com/gifs' ) !== -1 ) {
+			giphyID = this.splitAndLast( this.splitAndLast( searchText, '/' ), '-' );
+		}
+		// If search is hardcoded Giphy URL following this patterh: http://i.giphy.com/4ZFekt94LMhNK.gif
+		if ( searchText.indexOf( '//i.giphy.com' ) !== -1 ) {
+			giphyID = this.splitAndLast( searchText, '/' ).replace( '.gif', '' );
+		}
+		// https://media.giphy.com/media/gt0hYzKlMpfOg/giphy.gif
+		const match = searchText.match(
+			/http[s]?:\/\/media.giphy.com\/media\/([A-Za-z0-9\-\.]+)\/giphy.gif/
+		);
+		if ( match ) {
+			giphyID = match[ 1 ];
+		}
+		if ( giphyID ) {
+			return this.fetch( this.urlForId( giphyID ) );
+		}
+
+		return this.fetch( this.urlForSearch( searchText ) );
+	}, 250 );
+
+	urlForSearch = searchText => {
+		const { apiKey } = settings;
+		return `//api.giphy.com/v1/gifs/search?q=${ encodeURIComponent(
+			searchText
+		) }&api_key=${ apiKey }&limit=1`;
+	};
+
+	urlForId = giphyId => {
+		const { apiKey } = settings;
+		return `//api.giphy.com/v1/gifs/${ giphyId }?api_key=${ apiKey }`;
+	};
+
+	splitAndLast = ( array, delimiter ) => {
+		const split = array.split( delimiter );
+		return split[ split.length - 1 ];
+	};
+
+	fetch = url => {
+		const { setAttributes } = this.props;
+		const xhr = new XMLHttpRequest();
+		xhr.open( 'GET', url );
+		xhr.onload = () => {
+			if ( xhr.status === 200 ) {
+				const res = JSON.parse( xhr.responseText );
+				const giphyData = res.data.length > 0 ? res.data[ 0 ] : res.data;
+				// No results
+				if ( ! giphyData.images ) {
+					return;
+				}
+				const topPadding = Math.floor(
+					( giphyData.images.original.height / giphyData.images.original.width ) * 100
+				);
+				const giphyUrl = giphyData.embed_url;
+				setAttributes( { giphyUrl, topPadding } );
+				this.maintainFocus( 500 );
+			} else {
+				// Error handling TK
+			}
+		};
+		xhr.send();
+	};
+
+	setFocus = () => {
+		this.maintainFocus();
+		this.textControlRef.current.querySelector( 'input' ).focus();
+	};
+
+	maintainFocus = ( timeoutDuration = 3500 ) => {
+		this.setState( { focus: true }, () => {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+			if ( this.hasSearchText() ) {
+				this.timer = setTimeout( () => {
+					this.setState( { focus: false } );
+				}, timeoutDuration );
+			}
+		} );
+	};
+
+	clearFocus = () => {
+		this.setState( { focus: false }, () => {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+		} );
+	};
+
+	hasSearchText = () => {
+		const { attributes } = this.props;
+		const { searchText } = attributes;
+		return searchText && searchText.length > 0;
+	};
+
+	render() {
+		const { attributes, className, isSelected, setAttributes } = this.props;
+		const { align, caption, giphyUrl, searchText, topPadding } = attributes;
+		const { focus } = this.state;
+		const style = {
+			paddingTop: `${ topPadding }%`,
```

Comment:
It would be nice to just use the attribute without appending `%`, seems like we could store that complete string.

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/29638/commits/aefd179d872ea0c58e0de15d435bd68fa65e6167. Note that this commit corresponds to https://github.com/Automattic/jetpack/pull/10989/commits/181de07ce813cf0cad08be143c41f33eac8b39e4, and requires pulling current code from Jetpack repo so that Preview/Publish continues to work properly.